src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -204,138 +204,109 @@ protected override Conversion GetCollectionExpressionConversion(
                 }
             }
 
-            ImmutableArray<Conversion> elementConversions;
-            if (CollectionUsesKeyValuePairs(Compilation, collectionTypeKind, elementType, out var keyType, out var valueType))
+            var elementKeyValueTypes = TryGetCollectionKeyValuePairTypes(Compilation, collectionTypeKind, elementType);
+            var builder = ArrayBuilder<Conversion>.GetInstance(elements.Length);
+            foreach (var element in elements)
             {
-                var builder = ArrayBuilder<Conversion>.GetInstance(elements.Length * 2);
-                foreach (var element in elements)
+                if (element is BoundCollectionExpressionWithElement)
                 {
-                    switch (element)
+                    // Collection arguments do not affect convertibility.
+                    continue;
+                }
+                Conversion elementConversion = GetCollectionExpressionElementConversion(element, elementType, elementKeyValueTypes, ref useSiteInfo);
+                if (!elementConversion.Exists)
+                {
+                    builder.Free();
+                    return Conversion.NoConversion;
```

Comment:
Should we finish the loop so that we can collect all the use-site information from remaining elements?

Reply:
Since the conversion has failed, the caller will likely call `GenerateImplicitConversionErrorForCollectionExpression()` to report the specific failure, and that method should report all use-site diagnostics.